packages/vue-language-core/src/utils/localTypes.ts

```diff
@@ -64,8 +64,9 @@ export declare function makeOptional<T>(t: T): { [K in keyof T]?: T[K] };
 // TODO: make it stricter between class component type and functional component type
 export type ExtractComponentSlots<T> =
 	IsAny<T> extends true ? Record<string, any>
-	: T extends { ${slots}?: infer S } ? { [K in keyof S]-?: S[K] extends ((obj: infer O) => any) | undefined ? O : any }
-	: T extends { children?: infer S } ? { [K in keyof S]-?: S[K] extends ((obj: infer O) => any) | undefined ? O : any }
+	: T extends { ${slots}?: infer S } ? S
+	: T extends { children?: infer S } ? S
+	: T extends { [K in keyof PickNotAny<JSX.ElementChildrenAttribute, {}>]?: infer S } ? S
```

Comment:
- Shouldn't this be first? DefineComponent always has $slots but the default value is essentially `any`, so ElementChildrenAttribute will never be checked.
- `ElementChildrenAttribute` is a property on the elements' props. In vue that's `$props`, so `JSX.ElementAttributesProperty` also needs to be used. The slots will be on `T[keyof JSX.ElementAttributesProperty][keyof JSX.ElementChildrenAttribute]`